### PR TITLE
QL: Introduce Alias.unwrap method

### PR DESCRIPTION
**Pull request by costin**
Utility method for a common pattern through-out the code base -
 unwrapping a given expression if it's an Alias otherwise return it as
 is.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java [hunk 1]
```diff
@@ -1286,20 +1286,19 @@ public LogicalPlan apply(LogicalPlan plan) {
         private static LogicalPlan normalize(Aggregate aggregate, AttributeMap<Expression> aliases) {
             var aggs = aggregate.aggregates();
             List<NamedExpression> newAggs = new ArrayList<>(aggs.size());
-            boolean changed = false;
```
Comment (costin) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java:1289]: This change is bigger as it moves the replacing the agg/af manually to using transformDown instead.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java:1]: probably unintentional.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java [hunk 3]
```diff
@@ -149,16 +147,13 @@ private static void checkAggregate(LogicalPlan p, Set<Failure> failures) {
...
                 if (exp instanceof AggregateFunction af) {
                     af.field().forEachDown(AggregateFunction.class, f -> {
                         failures.add(fail(f, "nested aggregations [{}] not allowed inside other aggregations [{}]", f, af));
                     });
                 } else {
-                    if (Expressions.match(agg.groupings(), g -> {
-                        Expression to = g instanceof Alias al ? al.child() : g;
-                        return to.semanticEquals(exp);
-                    }) == false) {
+                    if (Expressions.match(agg.groupings(), g -> Alias.unwrap(g).semanticEquals(exp)) == false) {
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java:156]: 👌

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java [hunk 4]
```diff
@@ -82,11 +82,9 @@ else if (p.resolved()) {
...
             if (p instanceof Aggregate aggregate) {
                 for (NamedExpression agg : aggregate.aggregates()) {
-                    if (agg instanceof Alias as) {
-                        var child = as.child();
-                        if (child instanceof UnresolvedAttribute u) {
-                            failures.add(fail(child, "invalid stats declaration; [{}] is not an aggregate function", child.sourceText()));
-                        }
+                    var child = Alias.unwrap(agg);
+                    if (child instanceof UnresolvedAttribute u) {
+                        failures.add(fail(child, "invalid stats declaration; [{}] is not an aggregate function", child.sourceText()));
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java:87]: nit: `u` isn't used. But wondering if it couldn't be put to use:
```suggestion
                    if (Alias.unwrap(agg) instanceof UnresolvedAttribute u) {
                        failures.add(fail(u, "invalid stats declaration; [{}] is not an aggregate function", u.sourceText()));
```

**Review by bpintea — APPROVED** [review]
nice.

**Review by astefan — APPROVED** [review]
LGTM

**Comment by costin** [comment]
run elasticsearch-ci/part-3

**Comment by costin** [comment]
run elasticsearch-ci/part-3

**Pull request merged**